services/management/ia/templates_service.go

```diff
@@ -61,25 +69,42 @@ type TemplatesService struct {
 	l                 *logrus.Entry
 	userTemplatesPath string
 
+	host       string
+	publicKeys []string
+
 	rw        sync.RWMutex
 	templates map[string]templateInfo
 }
 
 // NewTemplatesService creates a new TemplatesService.
-func NewTemplatesService(db *reform.DB) *TemplatesService {
+func NewTemplatesService(db *reform.DB) (*TemplatesService, error) {
 	l := logrus.WithField("component", "management/ia/templates")
 
 	err := dir.CreateDataDir(templatesDir, "pmm", "pmm", dirPerm)
 	if err != nil {
 		l.Error(err)
 	}
 
-	return &TemplatesService{
+	host, err := envvars.GetSAASHost()
+	if err != nil {
+		return nil, err
+	}
+
+	s := &TemplatesService{
 		db:                db,
 		l:                 l,
 		userTemplatesPath: templatesDir,
+		host:              host,
 		templates:         make(map[string]templateInfo),
 	}
+
+	// TODO: same code exists in check service, move it to a better place.
+	if k := os.Getenv(envPublicKey); k != "" {
```

Comment:
What about envvar package?

Reply:
forgot about that, thanks, moved to envvars